interfaces/inc/sync.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef SYNC_H
+#define SYNC_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef _WIN32
+#ifndef volatile_atomic
+#define volatile_atomic volatile
+#else
+#define volatile_atomic volatile _Atomic
+#endif
+#endif
+
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout);

```

Comment:
>wait_on_address [](start = 26, length = 15)

It's a good idea to create a .c file with the header that includes this, has minimal stubs, and is actually built (in the cmakelist). That helps to catch bugs in the header.

Reply:
I guess two .c files in the case of PAL :)

---
In reply to: [441216567](https://github.com/Azure/azure-c-pal/pull/5#discussion_r441216567) [](ancestors = 441216567)